lib/readyset.rb

```diff
@@ -93,8 +91,43 @@ def self.drop_cache!(name_or_id: nil, sql: nil)
   #
   # @param [Array<Object>] *sql_array the SQL array to be executed against ReadySet
   # @return [PG::Result]
-  def self.raw_query(*sql_array)
-    ActiveRecord::Base.establish_connection(Readyset::Configuration.configuration.database_url)
-    ActiveRecord::Base.connection.execute(ActiveRecord::Base.sanitize_sql_array(sql_array))
+  def self.raw_query(*sql_array) # :nodoc:
+    ActiveRecord::Base.connected_to(role: reading_role, shard: shard, prevent_writes: false) do
+      ActiveRecord::Base.connection.execute(ActiveRecord::Base.sanitize_sql_array(sql_array))
+    end
+  end
+
+  # Routes to ReadySet any queries that occur in the given block. If `prevent_writes` is false, an
+  # attempt to execute a write within the given block will raise an error. Keep in mind that if
+  # `prevent_writes` is true, any writes that occur within the given block will be proxied through
```

Comment:
I think the usage of `prevent_writes` here might be flipped.


From the [Rails guides](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionHandling.html#method-i-connected_to_many):
> `reading` will automatically set prevent_writes to true
ie, `reads-only (true) = prevent_writes (true`

However, in the yard docs:
>    `prevent_writes` is true, any writes that occur within the given block will be proxied through Readyset to the database
I think suggests that it won't prevent the write from occurring. 


I might be missing some context with how the ReadySet server works though.

Reply:
Whoops, nice catch! 🤦